src/Stack/Runners.hs

```diff
@@ -261,10 +262,18 @@ munlockFile Nothing = return ()
 munlockFile (Just lk) = liftIO $ unlockFile lk
 
 -- Plumbing for --test and --bench flags
+-- Force enable --no-install-ghc and --skip-ghc-check flags
 withBuildConfigDot :: DotOpts -> GlobalOpts -> RIO EnvConfig () -> IO ()
-withBuildConfigDot opts go f = withBuildConfig go' f
+withBuildConfigDot opts go = withBuildConfig (updateGlobalOpts go)
   where
-    go' =
-        (if dotTestTargets opts then set (globalOptsBuildOptsMonoidL.buildOptsMonoidTestsL) (Just True) else id) $
-        (if dotBenchTargets opts then set (globalOptsBuildOptsMonoidL.buildOptsMonoidBenchmarksL) (Just True) else id)
-        go
+    updateGlobalOpts
+      = updateOpts (dotTestTargets opts)  (globalOptsBuildOptsMonoidL.buildOptsMonoidTestsL)      (Just True)
+      . updateOpts (dotBenchTargets opts) (globalOptsBuildOptsMonoidL.buildOptsMonoidBenchmarksL) (Just True)
+      . updateOpts True                   (globalOptsL.configMonoidSkipGHCCheckL)                 (Just True)
```

Comment:
It doesn't look like the `updateOpts` is really useful here, how about just `set (globalOptsL...) (Just True)`? I think it would be clearer that this is _always_ being set.

Reply:
Right!